catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -135,11 +152,87 @@ public void initialize(Map<String, String> conf) throws RuntimeException {
     // and gravitinoConfig will be passed to Hive config, and gravitinoConfig has higher priority
     mergeConfig.forEach(hadoopConf::set);
     hiveConf = new HiveConf(hadoopConf, HiveCatalogOperations.class);
+    UserGroupInformation.setConfiguration(hadoopConf);
+
+    initKerberosIfNecessary(conf, hadoopConf);
 
     this.clientPool =
         new CachedClientPool(getClientPoolSize(conf), hiveConf, getCacheEvictionInterval(conf));
   }
 
+  private void initKerberosIfNecessary(Map<String, String> conf, Configuration hadoopConf) {
+    if (UserGroupInformation.AuthenticationMethod.KERBEROS
+        == SecurityUtil.getAuthenticationMethod(hadoopConf)) {
+      try {
+        File keytabTemporaryDirectory = new File("tmp");
+        if (!keytabTemporaryDirectory.exists()) {
+          // Ignore the return value, because there exists many Hive catalog operations making
+          // this directory.
+          keytabTemporaryDirectory.mkdir();
+        }
+
+        // The id of entity is a random unique id.
+        File keytabFile = new File(String.format("tmp/gravitino-%s-keytab", entity.id()));
+        keytabFile.deleteOnExit();
+        if (keytabFile.exists() && !keytabFile.delete()) {
+          throw new IllegalStateException(
+              String.format("Fail to delete key tab file %s", keytabFile.getAbsolutePath()));
+        }
+
+        String keytabUri =
+            (String)
+                catalogPropertiesMetadata.getOrDefault(conf, HiveCatalogPropertiesMeta.KET_TAB_URI);
+        Preconditions.checkArgument(
+            StringUtils.isNotBlank(keytabUri), "If you use Kerberos, key tab uri can't be blank");
+        // TODO: Support to download the file from Kerberos HDFS
+        Preconditions.checkArgument(
+            !keytabUri.trim().startsWith("hdfs"), "Key tab uri doesn't support to use HDFS");
+
+        int fetchKeytabFileTimeout =
+            (int)
+                catalogPropertiesMetadata.getOrDefault(
+                    conf, HiveCatalogPropertiesMeta.FETCH_TIMEOUT_SEC);
+
+        FetchFileUtils.fetchFileFromUri(keytabUri, keytabFile, fetchKeytabFileTimeout, hadoopConf);
+
+        hiveConf.setVar(ConfVars.METASTORE_KERBEROS_KEYTAB_FILE, keytabFile.getAbsolutePath());
+
+        String catalogPrincipal = (String) catalogPropertiesMetadata.getOrDefault(conf, PRINCIPAL);
+        Preconditions.checkArgument(
+            StringUtils.isNotBlank(catalogPrincipal),
+            "If you use Kerberos, principal can't be blank");
```

Comment:
Please use the third-person statement here and above.

Reply:
Fixed.